src/validation/schema.js

```diff
@@ -322,13 +378,200 @@ const validateEntityFields = (defs, def) =>
     List(),
   )
 
+const validateNoImportsDirective = def =>
+  def.directives.find(directive => directive.name.value == 'imports')
+    ? List().push(
+        immutable.fromJS({
+          loc: def.name.loc,
+          entity: def.name.value,
+          message: `@imports directive only allowed on '${SUBGRAPH_SCHEMA}' type`,
+        }),
+      )
+    : List()
+
+const importDirectiveTypeValidators = {
+  StringValue: (_def, _directive, _type) => List(),
+  ObjectValue: (def, directive, type) => {
+    let errors = List()
+    if (type.fields.length != 2) {
+      return errors.push(
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: '@imports type objects accept and require two fields: [name, as]',
+        }),
+      )
+    }
+    return type.fields.reduce((errors, field) => {
+      if (!['name', 'as'].includes(field.name.value)) {
+        return errors.push(
+          immutable.fromJS({
+            loc: directive.name.loc,
+            entity: def.name.value,
+            message: `@imports type object field '${field.name.value}' invalid,  may only be one of: [name, as]`,
+          }),
+        )
+      }
+      if (field.value.kind != 'StringValue') {
+        return errors.push(
+          immutable.fromJS({
+            loc: directive.name.loc,
+            entity: def.name.value,
+            message: '@imports type object fields [name, as] must be strings',
+          }),
+        )
+      }
+      return errors
+    }, errors)
+  },
+}
+
+const validateImportDirectiveType = (def, directive, type) =>
+  importDirectiveTypeValidators[type.kind]
+    ? importDirectiveTypeValidators[type.kind](def, directive, type)
+    : List().push(
+        immutable.fromJS({
+          loc: directive.name.loc,
+          entity: def.name.value,
+          message: 'Imported type must be either String | {name: String, as: String}',
+        }),
+      )
+
+const validateImportDirectiveArgumentTypesIsValid = (def, directive, argument) => {
+  if (argument.value.kind != 'ListValue') {
+    return List().push(
+      immutable.fromJS({
+        loc: directive.name.loc,
+        entity: def.name.value,
+        message: '@imports directive argument: types must be an list',
```

Comment:
Let's `types` in backticks or quotes to be consistent. It would be good to make the errors consistent across the board in the changes.